### DISCUSSION TITLE
Remove Cell Outputs ??

### DISCUSSION THREAD

**Discussion by gzt5142** [hytest-org/hytest#255] [Repo Admin]
Related to #254 

Regarding the practice of editing notebooks outside of their native environment ....  The stream editor (`sed`) and other low level tools will operate on the JSON document as a whole. That's powerful and useful, but should be done with some care.  It is possible to damage a notebook if some of the necessary JSON delimiters are mangled (dropping a `}` or a `,` for example). 

Relatedly, the cell outputs are worth considering -- in specific, it might be worthwhile to standardize on the practice of **removing** cell outputs as a part of the git process. This would remove them from complicating bulk edits, and would also reduce the size of the notebook overall (saving storage space, network bandwidth, download time, etc). 

I really like `nbstripout` for this (installed as a git [filter](https://github.com/gzt5142/gzt5142/blob/main/git-isms/Filters.md) ), and use it on my other(non-HyTEST) projects. Other solutions exist and might be worth considering as a part of the recommended workflow for HyTEST contributors.  It could also be a little over-kill, as it ups the technical pre-requisites for your contributors.

**Comment by alaws-USGS** [comment]
I think this is a great idea, especially as cell outputs make it more difficult to easily preview notebooks on GH.
